mdp/nodes/misc_nodes.py

```diff
@@ -815,3 +826,603 @@ def _execute(self, x):
+    The node also provides a utility method:
+     'get_neighbors' - Returns the neighboring
+     grid points based on a distance function applied to
+     the graph coordinates. The distance function
+     can be specified using the 'nbr_dist_fn' argument
+     and the threshold can be specified using the
+     'nbr_radius' argument.
+
+     'get_adjacency_matrix' - Returns an adjacency matrix
+     of the graph.
+
+    """
+
+    def __init__(self, grid_lims, n_grid_pts=None, output_type=None, nbr_dist_fn=1, nbr_radius=1,
+                 include_central_elem=True, input_dim=None, dtype=None):
+        """
+        grid_lims - a tuple of lower and upper bounds for each dimension of the input grid.
+                Eg., [(lower1, lower2, ...,), (upper1, upper2, ...,)]
+
+        n_grid_pts - number of grid points within each dimension. This provides the spacing between
+                the grid points. Default is set to the number of integers between the bounds.
+
+        output_type - the type of conversion. Accepted values:
+                    'gridx' - default (just returns the input)
+                    'graphx' - returns graph coordinates
+                    'graphindx' - returns graph indices of the input.
+
+        nbr_dist_fn - distance function to define the closest neighbors (applied on the graph
+                    coordinates). This is the same as
+                    "order" argument of the numx_linalg function. (Doc added here for quick reference)
+                     Accepted values:
+                    {non-zero int, inf, -inf, 'fro', 'nuc'}, inf means numpy's `inf` object.
+
+                    For values of ``nbr_dist_fn <= 0``, the result is, strictly speaking, not a
+                    mathematical 'norm', but it may still be useful for various numerical
+                    purposes.
+
+                    The following norms can be calculated:
+
+                ==========  ============================  ==========================
+                nbr_dist_fn    norm for matrices             norm for vectors
+                ==========  ============================  ==========================
+                    None   Frobenius norm                2-norm
+                    'fro'  Frobenius norm                --
+                    'nuc'  nuclear norm                  --
+                    inf    max(sum(abs(x), axis=1))      max(abs(x))
+                    -inf   min(sum(abs(x), axis=1))      min(abs(x))
+                    0      --                            sum(x != 0)
+                    1      max(sum(abs(x), axis=0))      as below
+                    -1     min(sum(abs(x), axis=0))      as below
+                    2      2-norm (largest sing. value)  as below
+                    -2     smallest singular value       as below
+                    other  --                            sum(abs(x)**ord)**(1./ord)
+                    =====  ============================  ==========================
```

Comment:
Maybe one could use an enum or something like that (we would have to use enum34 for backwards compatibility). This might also be another chance for the strategy pattern (e.g., with a factory function for l-norm functions)